### ISSUE TITLE
ndi.binary_closing weird behavior on different WSL distributions.

### ISSUE THREAD

**Issue by Jarartur** [cupy/cupy#8445]
### Description

Hello,
I noticed very weird behavior on my two WSL distributions: Ubuntu and Pengwin.

On Pengwin my code runs in about 3 seconds. On Ubuntu it's 44 seconds. Keep in mind that this is the same hardware, some cupy versions, same cuda driver, etc.

I tested the outputs of the code on a few different images (it's an image processing script) and both systems produce the same results. How is that possible?

The scripts are mainly binary morphology, labeling and edt. Unfortunately, I cannot share it whole, but I attach the benchmarking I've run. 

I do not really know where to begin searching for the answer. What is weirder, the Pengwin one does not even register in nsys profiler. But I do see in task manager that GPU utilization jumps to 100%.

I tested the code on a computing server on A100 GPU and on a rtx2080. Both of them also achieved about 44 seconds.

### To Reproduce

```py
# import numpy as np
import cupy as np
import SimpleITK as sitk
# import scipy.ndimage as ndi
import cupyx.scipy.ndimage as ndi

from cupyx.profiler import benchmark

# 3D image
img = np.array(sitk.GetArrayFromImage(sitk.ReadImage('anatomical/caseid.mha')))
img = img>0

def img_dilation(img):
    return ndi.binary_closing(img, structure=ndi.generate_binary_structure(3, 1), iterations=20, brute_force=True)

def img_labeling(img):
    return ndi.label(img, structure=ndi.generate_binary_structure(3, 1))

def img_edt(img):
    return ndi.distance_transform_edt(~img, return_distances=False, return_indices=True)

print(benchmark(img_dilation, (img,), n_repeat=20))
print(benchmark(img_labeling, (img,), n_repeat=20))
print(benchmark(img_edt, (img,), n_repeat=20))
```

### Installation

Conda-Forge (`conda install ...`)

### Environment

Ubuntu:
```
OS                           : Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python Version               : 3.10.14
CuPy Version                 : 13.2.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 2.0.1
SciPy Version                : None
Cython Build Version         : 0.29.37
Cython Runtime Version       : None
CUDA Root                    : None
nvcc PATH                    : None
CUDA Build Version           : 12050
CUDA Driver Version          : 12030
CUDA Runtime Version         : 12050 (linked to CuPy) / RuntimeError('CuPy failed to load libcudart.so.12: OSError: libcudart.so.12: cannot open shared object file: No such file or directory') (locally installed)
cuBLAS Version               : (available)
cuFFT Version                : 11203
cuRAND Version               : 10306
cuSOLVER Version             : (11, 6, 3)
cuSPARSE Version             : (available)
NVRTC Version                : (12, 5)
Thrust Version               : 200400
CUB Build Version            : 200200
Jitify Build Version         : <unknown>
cuDNN Build Version          : 8907
cuDNN Version                : 8907
NCCL Build Version           : 22203
NCCL Runtime Version         : 22203
cuTENSOR Version             : 20001
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA GeForce RTX 3080 Laptop GPU
Device 0 Compute Capability  : 86
Device 0 PCI Bus ID          : 0000:01:00.0
```

Pengwin:
```
OS                           : Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.36
Python Version               : 3.10.14
CuPy Version                 : 13.2.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 2.0.1
SciPy Version                : None
Cython Build Version         : 0.29.37
Cython Runtime Version       : None
CUDA Root                    : /usr/local/cuda
nvcc PATH                    : /usr/local/cuda/bin/nvcc
CUDA Build Version           : 12050
CUDA Driver Version          : 12030
CUDA Runtime Version         : 12050 (linked to CuPy) / 12030 (locally installed)
cuBLAS Version               : (available)
cuFFT Version                : 11203
cuRAND Version               : 10306
cuSOLVER Version             : (11, 6, 3)
cuSPARSE Version             : (available)
NVRTC Version                : (12, 5)
Thrust Version               : 200400
CUB Build Version            : 200200
Jitify Build Version         : <unknown>
cuDNN Build Version          : None
cuDNN Version                : None
NCCL Build Version           : None
NCCL Runtime Version         : None
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA GeForce RTX 3080 Laptop GPU
Device 0 Compute Capability  : 86
Device 0 PCI Bus ID          : 0000:01:00.0
```

### Additional Information

Ubuntu is a fresh install. Pengwin was used for a moment for pytorch's c++ api so there is probably something installed.

**Comment by Jarartur** [comment]
The benchmark timings:

Ubuntu:
```
img_dilation        :    CPU: 2051688.381 us   +/- 11219.377 (min: 2042631.862 / max: 2096096.722) us     GPU-0: 2051693.909 us   +/- 11217.155 (min: 2042630.005 / max: 2096079.590) us
img_labeling        :    CPU:  4426.872 us   +/- 344.029 (min:  4031.118 / max:  5300.526) us     GPU-0:  6015.214 us   +/- 335.215 (min:  5495.712 / max:  6736.896) us
img_edt             :    CPU:   731.037 us   +/- 147.880 (min:   502.126 / max:  1168.927) us     GPU-0: 27338.920 us   +/- 315.147 (min: 27040.768 / max: 28371.967) us
```

Pengwin:
```
img_dilation        :    CPU: 149308.858 us   +/- 984.077 (min: 147266.905 / max: 150941.184) us     GPU-0: 149329.196 us   +/- 984.914 (min: 147302.139 / max: 150962.463) us
img_labeling        :    CPU:  3981.388 us   +/- 59.228 (min:  3869.188 / max:  4113.540) us     GPU-0:  5445.570 us   +/- 55.062 (min:  5339.136 / max:  5571.584) us
img_edt             :    CPU:   738.259 us   +/- 99.856 (min:   631.044 / max:  1049.825) us     GPU-0: 27389.986 us   +/- 344.256 (min: 27121.664 / max: 28656.639) us
```

**Comment by Jarartur** [comment]
Well, once I looked again at this bug report I figured it out myself. I am gonna leave this for anyone like me:
the difference was the nvcc compilator. I had one in Pengwin distro from Pytorch C++ days and not in Ubuntu.

Maybe that is a suggestion for docs, because I thought that when installing from conda I get everything bundled. Turns out for best performance you still need cuda toolkit installed locally.

I know the documentation claims this on top of the [page](https://docs.cupy.dev/en/stable/install.html#installing-cupy-from-conda-forge) but I though it only applies to pip installation.

So all in all I am blaming me 😅.
